doc/microsite/src/main/tut/distage/distage-framework-docker.md

```diff
@@ -19,3 +20,487 @@ libraryDependencies += "io.7mind.izumi" %% "distage-framework-docker" % "$izumi.
+
+### Usage in Integration Tests
+
+A common use case is using docker containers to provide service implementations for integration test.
+Such as using a postgres container for verifying an application that uses a postgres database.
+Distage container resources are easy to integrate as providers.
+
+```scala mdoc:silent
+// the postgres configuration used by the application.
+case class PostgresServerConfig(
+  host: String,
+  port: Int
+)
+
+// a module that binds the implementation of PostgresServerConfig
+// to one provided by a docker container
+class PostgresServerConfigDockerModule[F[_]: TagK] extends ModuleDef {
+  make[PostgresServerConfig].from {
+    container: PostgresDocker.Container => {
+      val knownAddress = container.availablePorts.availablePorts(PostgresDocker.primaryPort).head
+      PostgresServerConfig(knownAddress.hostV4, knownAddress.port)
+    }
+  }
+
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[F]
+  }
+}
+```
+
+Suppose our application uses [skunk](https://tpolecat.github.io/skunk) for postgres access
+
+```scala mdoc:silent
+import skunk.implicits._
+import skunk.codec.all._
+
+class SkunkPostgresExample(session: skunk.Session[IO]) {
+  val run: IO[String] = for {
+    serverDate <- session.unique(sql"select current_date".query(date))
+    _ <- IO(println(s"current server date is ${serverDate}."))
+  } yield serverDate.toString
+}
+```
+
+An integration test would use a module that provides the `Session[IO]` using
+`PostgresDocker.Container`:
+
+```scala mdoc:silent
+import natchez.Trace.Implicits._
+import scala.concurrent.ExecutionContext.Implicits.global
+
+implicit val contextShift = IO.contextShift(global)
+implicit val timer = IO.timer(global)
+
+class SkunkSessionUsingDockerModule extends ModuleDef {
+  make[skunk.Session[IO]].fromResource {
+    container: PostgresDocker.Container => {
+      val knownAddress = container.availablePorts.availablePorts(PostgresDocker.primaryPort).head
+      skunk.Session.single[IO](
+        host = knownAddress.hostV4,
+        port = knownAddress.port,
+        user = "postgres",
+        password = Some("postgres"),
+        database = "postgres"
+      )
+    }
+  }
+
+  make[PostgresDocker.Container].fromResource {
+    PostgresDocker.make[IO]
+  }
+}
+```
+
+Typically, this would be used in a test runner such as `DistageSpecScalatest[IO]`. For completeness,
+the example can be run directly using:
+
+```scala mdoc:to-string
+def postgresIntegrationExample = {
+  val applicationModules = new ModuleDef {
+    include(new SkunkSessionUsingDockerModule)
+    include(distageFrameworkModules)
+    make[SkunkPostgresExample]
+  }
+
+  Injector().produceGetF[IO, SkunkPostgresExample](applicationModules).use { app =>
+    app.run
+  }
+}
+
+println(postgresIntegrationExample.unsafeRunSync())
+```
+
+### Docker Container Environment
+
+The container config (@scaladoc[Docker.ContainerConfig](izumi.distage.docker.Docker.ContainerConfig)
```

Comment:
Does not resolve, may be link to `izumi.distage.docker.Docker$$ContainerConfig` ?